src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Interop/LegacyExports.cs

```diff
@@ -210,13 +211,24 @@ public static void CreateDateTimeRef(double ticks, out object result)
             result = unixTime.DateTime;
         }
 
+        // we do this via reflection to allow linker to trim dependency on Uri class and it's assembly
+        // if the user code has methods with Uri signature, they probably also have the Uri constructor
+        // if they don't have it, they could configure ILLing to protect it after they enabled trimming
+        // We believe that this code path is probably not even used in the wild
+        // System.Private.Uri is ~80KB large assembly so it's worth trimming
+        private static Type? uriType;
+
         [MethodImplAttribute(MethodImplOptions.NoInlining)] // https://github.com/dotnet/runtime/issues/71425
+        [Diagnostics.CodeAnalysis.UnconditionalSuppressMessage("Trimming", "IL2077", Justification = "Done on purpose, see comment above.")]
+        [Diagnostics.CodeAnalysis.UnconditionalSuppressMessage("Trimming", "IL2057", Justification = "Done on purpose, see comment above.")]
         public static void CreateUriRef(string uri, out object? result)
         {
-            // we do this via reflection to allow linker to trim dependency on URI and it's assembly
-            // if the user code has methods with Uri signature, this should work too
-            // System.Private.Uri is large assembly so it's worth trimming
-            var uriType = Type.GetType("System.Uri, System.Private.Uri");
+            if (uriType == null)
+            {
+                // StringBuilder to confuse ILLink, which is too smart otherwise
+                StringBuilder sb = new StringBuilder("System.Uri, System.Private.Uri");
+                uriType = Type.GetType(sb.ToString());
+            }
             if (uriType == null) throw new InvalidProgramException();
```

Comment:
Can we explain here what happend and what to do to fix it?
Will it make sense to use `ValueStringBuilder` to skip even the single allocation here?

Reply:
Added better exception message and sample configuration for not trimming it.